jni/include/knn_extension/faiss/MultiVectorResultCollector.h

```diff
@@ -0,0 +1,41 @@
+/*
+ * Copyright OpenSearch Contributors
+ * SPDX-License-Identifier: Apache-2.0
+ */
+
+#pragma once
+
+#include <faiss/impl/ResultCollector.h>
+#include <faiss/MetricType.h>
+#include "knn_extension/faiss/utils/BitSet.h"
+#include <unordered_map>
+
+namespace os_faiss {
+
+using idx_t = faiss::idx_t;
+/**
+ * Implementation of ResultCollector to support multi vector
+ *
+ * By using parent_bit_set, it convert a doc id to its parent doc id and store the parend doc id
+ * while collecting search result. Using group_id_to_index, it de-duplicates result from same parent
+ * doc. Once all results are collected, post_process method is called where it converts parent doc id
+ * to its original id using group_id_to_id.
+ */
+struct MultiVectorResultCollector:faiss::ResultCollector {
+    std::unordered_map<idx_t, idx_t> group_id_to_id;
+    std::unordered_map<idx_t, size_t> group_id_to_index;
+    BitSet* parent_bit_set;
```

Comment:
Who owns alloc/dealloc of this?

Reply:
The caller will handle it. In this case, it will be faiss_wrapper.
faiss_wrapper -> MultiVectorResultCollectorFactory -> MutiVectorResultCollector